### ISSUE TITLE
Change character set comparison function in `sort-alternatives`?

### ISSUE THREAD

**Issue by RunDevelopment** [ota-meshi/eslint-plugin-regexp#420]
Since we need to be able to sort character sets in `sort-alternatives`, I implemented a comparison function in #320. However, I recently discovered that this function doesn't sort that way you'd expect.

Example: `sort-alternatives` wants to swap the following alternatives:
```js
var re = /\b(?:sample|(?:un)?stable)\b/u;
```

This is quite strange to me because the first characters of `sample` and `(?:un)?stable` are `[s]` and `[su]` respectively, and I think `[s]` should come before the character set `[su]`.

Honestly, I don't know whether this is a bug or not. We don't specify *how* alternatives are sorted, so *technically* any order is correct. What do you think @ota-meshi?

**Comment by ota-meshi** [comment]
> I think [s] should come before the character set [su].

I agree with you :+1:

**Comment by RunDevelopment** [comment]
Resolved by #423.
